lambda/matching.mli

```diff
@@ -19,6 +19,11 @@ open Typedtree
 open Lambda
 open Debuginfo.Scoped_location
 
+type error =
```

Comment:
Do these declarations need to be here? I think not.

Reply:
In every case I've looked at, the types are in the mli file because some other module needs to get at them. That's not the case here.

Reply:
OK, removed.